data_write/data_write.py

```diff
@@ -327,10 +321,37 @@ def parse_antenna_iq(self):
                         antenna_iq_stage[ant_str] = {}
 
                     if 'data' not in antenna_iq_stage[ant_str]:
-                        antenna_iq_stage[ant_str]['data'] = antennas_data[ant_num, :]
-                    else:
-                        arr = antenna_iq_stage[ant_str]
-                        arr['data'] = np.concatenate((arr['data'], antennas_data[ant_num, :]))
+                        antenna_iq_stage[ant_str]['data'] = []
+                    antenna_iq_stage[ant_str]['data'].append(antennas_data[ant_num, :])
+
+    def numpify_arrays(self):
+        """ Consolidates data for each data type to one array.
+
+        In parse_[type](), new data arrays are appended to a list for speed considerations.
+        This function converts these lists into numpy arrays.
+        """
+        for slice_id, slice_data in self._antenna_iq_accumulator.items():
+            if isinstance(slice_id, int):       # filtering out 'data_descriptors'
+                for param_name, param_data in slice_data.items():
+                    for array_name, array_data in param_data.items():
+                        if array_name != 'num_samps':
+                            array_data['data'] = np.array(array_data['data'], dtype=np.complex64)
+
+        for slice_id, slice_data in self._bfiq_accumulator.items():
+            if isinstance(slice_id, int):       # filtering out 'data_descriptors'
+                for param_name, param_data in slice_data.items():
+                    if 'shm' in param_name: # not a dictionary, just the name of a shared memory object
```

Comment:
is this required anymore now that you're just copying data and closing shm immediately?

Reply:
It is not, good catch.